### PR TITLE
Simplify edx-pattern-library Sass Compilation

### PR DESCRIPTION
**Pull request by talbs**
This work focuses on simplifying edx-pattern-library sass compilation syntax (similar to how its done in [Bootstrap Sass](https://github.com/twbs/bootstrap-sass#sass-1) to just one line when including it in an app's compilation. This involves:

* creating the ``_edx-pattern-library.scss`` partial to import all lib partials
* adding in per-app (pldoc) configuration overrides
* renaming/syncing comments and style headings

- - -

##### TODOs

* [x] Sync up Wiki and README Compilation code examples

- - -

##### Reviewers

* [x] FED + Sass architecture - @frrrances

### CODE REVIEW

**Comment by talbs** [comment]
@frrrances, mind re-reviewing? I added some more simplification that let's anyone **call the edx-pattern-library in 1 Sass import**. Check out - https://github.com/edx/ux-pattern-library/commit/fc798d518e326c482746f69d8dde85880dc21c74.

In short, this:

* moves and third party library references that ``edx-pattern-library`` needs to the ``_edx-pattern-library.scss`` partial
* sets a ``layout-direction`` Sass var in the ``*-ltr/rtl.scss`` files
* that directional variable is then passed into the ``_edx-pattern-library.scss`` partial and any directionally-conditional third party libs (bi-app-sass) and variables (susy's grid-direction) are passed into the rest of the compile.

- - -

This means: 

* no more multiple paths to mess up when trying to add the edx-pattern-library to your app's Sass compile
* all libs are properly loaded with the edx-pattern-library automatically
* import paths are simplified once in the ``_edx-pattern-library.scss`` partial to just ``utilities/**/**``

- - -

I'm updating doc to accompany this now. FYI, @clrux and @marcotuts

File: _src/pldoc/static/sass/main-rtl.scss [hunk 1]
```diff
@@ -6,17 +6,9 @@
 // NOTE: This is the left to right (RTL) configured style compile. It should mirror main-rtl w/ the exception of bi-app references.
```
Comment (frrrances) [inline _src/pldoc/static/sass/main-rtl.scss]: Note not needed anymore?

File: _src/pldoc/static/sass/main-ltr.scss [hunk 2]
```diff
@@ -6,17 +6,9 @@
 // NOTE: This is the left to right (LTR) configured style compile. It should mirror main-rtl w/ the exception of bi-app references.
```
Comment (frrrances) [inline _src/pldoc/static/sass/main-ltr.scss]: Note not needed anymore?

File: _src/pldoc/static/sass/main-ltr.scss [hunk 3]
```diff
@@ -6,17 +6,9 @@
...
 
 // ------------------------------
-// #VENDOR
+// #CONFIG - layout direction
```
Comment (frrrances) [inline _src/pldoc/static/sass/main-ltr.scss]: Are there other config things that might go in this section?

**Comment by frrrances** [comment]
Totally sweet cleanup! Couple small comments, but otherwise :+1: :shipit:

File: _src/pldoc/static/sass/main-ltr.scss (hunk 3 above)
Comment (talbs) [inline _src/pldoc/static/sass/main-ltr.scss]: Not in this section, but in the ``build.scss`` file, there is a place for the app's ``_config.scss`` which can add/replace variables, functions, mixins.

**Comment by talbs** [comment]
Thanks, @frrrances. Removed the notes. Merging after I update the wiki docs.

**Pull request merged**